packages/grafana-ui/src/types/icon.ts

```diff
@@ -1,9 +1,23 @@
+import React from 'react';
 import { ComponentSize } from './size';
-export type IconType = 'mono' | 'default';
+
+// Exports of main Icon types
+export type IconProps = SolidIconProps | DefaultIconProps;
```

Comment:
I wonder if it would make sense to combine solid and default icon props into one type 🤔

Reply:
So what I was trying to achieve is only suggest solid icon names when type="solid" and default icon names when type="default" or no type specified.

Reply:
I'm not sure this is possible to achieve with the current approach as TS needs to know the types before runtime. How about introducing a wrapper component that accepts `type` prop and then renders icon of the needed type? It does introduce one extra level of components, but that's a tradeoff for being able to switch types. Also the `Icon` itself will need to be adjusted to accept generic types:
```jsx
type IconType = DefaultIconType | SolidIconType | 'mono'; // mono needs to be specified on the type if we want to accept those;
interface IconProps<T> extends React.HTMLAttributes<HTMLDivElement> {
  name: T;
  type?: IconType;
  size?: IconSize;
}

const IconWrapper: React.FC<IconProps<IconName>> = ({ type = 'default', ...rest }) => {
  if (type === 'mono') {
    console.warn('mono type is deprecated, please use default or solid types');
  }
  return ['default', 'mono'].includes(type) ? IconRef<DefaultIconName>(rest) : IconRef<SolidIconName>(rest);
};

const IconBase: React.FC<IconProps<IconName>> = (
  { size = 'md', type = 'default', name, className, style, ...divElementProps },
  ref: React.Ref<HTMLDivElement>
) => {
  //...
};

IconWrapper.displayName = 'Icon';

export const IconRef = React.forwardRef(IconBase) as <T = IconName>(
  p: IconProps<T> & { ref?: React.Ref<HTMLDivElement> }
) => React.ReactElement;

export { IconWrapper as Icon };
```
Not 100% sure about this approach and typing `forwardRef` wrapped components is complicated, but [this answer](https://stackoverflow.com/a/58473012) was really helpful for me.

Reply:
I like @Clarity-89 idea. Honestly, I always struggled understanding the concept behind mono/default/solid. 

What I would expect as a consumer of the API are two icon components:

- `<Icon name="..." size="..." />` - default, line only icon
- `<SolidIcon name="..." size="..." />` _or_ `<Icon name="..." side="..." solid />` - solid icon

I'm not sure I'm missing anything here, but IMO the icon type property should be hidden from the consumer.